Yosemite/Yosemite/Stores/MetaDataStore.swift

```diff
@@ -0,0 +1,209 @@
+            updateProductMetaData(siteID: siteID, productID: productID, metadata: metadata, onCompletion: onCompletion)
+        }
+    }
+}
+
+// MARK: - Upsert MetaData for Orders and Products
+//
+private extension MetaDataStore {
+    /// Updates order metadata in the local database.
+    /// - Parameters:
+    ///   - siteID: Site id of the order.
+    ///   - orderID: ID of the order.
+    ///   - metadata: Metadata to be updated.
+    ///   - onCompletion: Callback called when the action is finished, including the result of the update operation.
+    ///
+    func updateOrderMetaData(siteID: Int64,
+                             orderID: Int64,
+                             metadata: [[String: Any]],
+                             onCompletion: @escaping (Result<[MetaData], Error>) -> Void) {
+        Task { @MainActor in
+            do {
+                let results = try await self.remote.updateMetaData(for: siteID, for: orderID, type: .order, metadata: metadata)
+                upsertStoredOrderMetaDataInBackground(readOnlyOrderMetaDatas: results, orderID: orderID, siteID: siteID) {
+                    onCompletion(.success(results))
+                }
+            } catch {
+                onCompletion(.failure(error))
+            }
+        }
+    }
+
+    /// Updates (OR Inserts) the specified ReadOnly OrderNote Entities into the Storage Layer.
+    /// - Parameters:
+    ///   - readOnlyOrderMetaDatas: Array of read-only order metadata.
+    ///   - orderID: ID of the order.
+    ///   - siteID: Site id of the order.
+    ///   - onCompletion: Callback called when the action is finished.
+    ///
+    func upsertStoredOrderMetaDataInBackground(readOnlyOrderMetaDatas: [Networking.MetaData], 
+                                               orderID: Int64,
+                                               siteID: Int64,
+                                               onCompletion: @escaping () -> Void) {
+        let derivedStorage = sharedDerivedStorage
+        derivedStorage.perform {
+            for readOnlyOrderMetaData in readOnlyOrderMetaDatas {
+                self.saveMetaData(derivedStorage, readOnlyOrderMetaData, orderID: orderID, siteID: siteID)
+            }
+        }
+
+        storageManager.saveDerivedType(derivedStorage: derivedStorage) {
+            DispatchQueue.main.async(execute: onCompletion)
+        }
+    }
+
+    /// Using the provided StorageType, update or insert a Storage.MetaData using the provided ReadOnly
+    /// MetaData. This func does *not* persist any unsaved changes to storage.
+    /// - Parameters:
+    ///   - storage: The storage type to use.
+    ///   - readOnlyMetaData: The read-only metadata to save.
+    ///   - orderID: ID of the order.
+    ///   - siteID: Site id of the order.
+    ///
+    private func saveMetaData(_ storage: StorageType, _ readOnlyMetaData: MetaData, orderID: Int64, siteID: Int64) {
```

Comment:
The `private` is not needed as this is already inside a private extension.

Reply:
Save as above 👍 https://github.com/woocommerce/woocommerce-ios/commit/a7e0b1ffb4494fc1f30eaa5431493ece830bb79a